tests/udp-server/main.go

```diff
@@ -0,0 +1,50 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"net"
+)
+
+func main() {
+	port := flag.String("port", "3334", "The port the server listens to")
+	flag.Parse()
+
+	listener, err := net.ListenPacket("udp", fmt.Sprintf(":%v", *port))
+	if err != nil {
+		log.Panicln(err)
+	}
+	defer listener.Close()
+	log.Printf("listening to udp connections at: :%v\n", *port)
+
+	buffer := make([]byte, 1024)
+
+	for {
+		n, addr, err := listener.ReadFrom(buffer)
+		if err != nil {
+			log.Panicln(err)
+		}
+
+		fmt.Printf("packet-received: bytes=%d from=%s\n", n, addr.String())
+
+		address := fmt.Sprintf("%v:%v", GetOutboundIP().String(), *port)
+		log.Printf("write data to connection: %v\n", address)
+
+		n, err = listener.WriteTo([]byte(address), addr)
+		if err != nil {
+			log.Panicln(err)
+		}
+		fmt.Printf("packet-written: bytes=%d to=%s\n", n, addr.String())
+	}
+}
+
+func GetOutboundIP() net.IP {
+	conn, err := net.Dial("udp", "8.8.8.8:80")
```

Comment:
I wonder if this can be source of any failures, considering also that it is called on every request handling.

If it can be, there is another way to get pod IP information - propagate it using a cli argument or env variable. https://kubernetes.io/docs/tasks/inject-data-application/environment-variable-expose-pod-information/#use-pod-fields-as-values-for-environment-variables  and https://github.com/nginxinc/kubernetes-ingress/blob/master/deployments/deployment/nginx-ingress.yaml#L57

Reply:
> I've reduced it to one call at the start of the service, to reduce the likelihood it fails.
I don't think the pod fields as values solution will work. We need the IP address which is not defined in the spec/metadata as far as I know.


I suggest trying that busybox example from https://kubernetes.io/docs/tasks/inject-data-application/environment-variable-expose-pod-information/#use-pod-fields-as-values-for-environment-variables - you would see that the env variable `MY_POD_IP` will have the pod IP. That env variable can be used in application arg like  `... -ip=$(MY_POD_IP)`

> I'd rather try this for now and if we get failures in the pipeline, then we can invest more time ?

this makes sense

Reply:
I dug a little deeper, turns out the pod ip is available. I've updated the PR with POD_IP as an env var.